### PR TITLE
[Enhancement](datetimev2-enhance) support 'microseconds_add' function for datetimev2

### LINKED ISSUE #16967: [Enhancement] Support function `microseconds_add` for datetimev2
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Description

As descriptions in #16561 
Now Doris already supports DATETIMEV2 which could have precision up to microseconds. But function microseconds_add is not supported yet

`CREATE TABLE datetime_table (date datetime(3) NOT NULL COMMENT '数据灌入日期时间' ) distributed by hash(date) buckets 1 properties("replication_num"="1");`

image

### Solution

1. Introduce new TimeUnit `MICROSECONDS` to `vdatetime_value`
2. `date_add_interval` computation should handle `MICROSECONDS` unit

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by WinkerDu**
# Proposed changes

Issue Number: close #16967 

## Problem summary

Describe your changes.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [x] Has unit tests been added
* [x] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by WinkerDu** [comment]
@Gabriel39 @yangzhg

**Comment by WinkerDu** [comment]
run buildall

**Comment by WinkerDu** [comment]
@jackwener @morrySnow PTAL, thank you.

**Comment by jackwener** [comment]
cc @BiteTheDDDDt

File: be/src/vec/runtime/vdatetime_value.cpp [hunk 1]
```diff
@@ -2553,17 +2553,25 @@ bool DateV2Value<T>::date_add_interval(const TimeInterval& interval, DateV2Value
...
-    if constexpr ((unit == SECOND) || (unit == MINUTE) || (unit == HOUR) ||
+    if constexpr ((unit == MICROSECOND) || (unit == SECOND) || (unit == MINUTE) || (unit == HOUR) ||
                   (unit == SECOND_MICROSECOND) || (unit == MINUTE_MICROSECOND) ||
                   (unit == MINUTE_SECOND) || (unit == HOUR_MICROSECOND) || (unit == HOUR_SECOND) ||
                   (unit == HOUR_MINUTE) || (unit == DAY_MICROSECOND) || (unit == DAY_SECOND) ||
                   (unit == DAY_MINUTE) || (unit == DAY_HOUR) || (unit == DAY) || (unit == WEEK)) {
         // This may change the day information
+        int64_t microseconds = this->microsecond() + sign * interval.microsecond;
+        int64_t extra_second = microseconds / 1000000L;
+        microseconds %= 1000000L;
```
Comment (BiteTheDDDDt) [inline be/src/vec/runtime/vdatetime_value.cpp:2564]: better define 1000000 to a constexpr value as a common constant.

Comment (BiteTheDDDDt) [reply be/src/vec/runtime/vdatetime_value.cpp:2564]: here can use microseconds-=extra_second*1000000, it is more fast than mod operator

Comment (WinkerDu) [reply be/src/vec/runtime/vdatetime_value.cpp]: OK, will fix it.

**Comment by WinkerDu** [comment]
run buildall

**Comment by WinkerDu** [comment]
@BiteTheDDDDt PTAL, thank you

**Review by BiteTheDDDDt — APPROVED** [review]
LGTM

**Comment by WinkerDu** [comment]
@BiteTheDDDDt Is this pr ok to merge? Or may you invite some other committers review? Thank you :)

**Pull request merged**